app/Console/Commands/OneOff/MigrateLocationCodes.php

```diff
@@ -0,0 +1,239 @@
+        ProjectPitch::whereNot('level_2_proposed', null)->update(['level_2_proposed' => null]);
+
+        $this->info("\n\nUpdating form question option lists...");
+        FormQuestion::where(['options_list' => 'countries', 'input_type' => 'select'])->update(['options_list' => 'gadm-level-0']);
+        FormQuestion::where(['options_list' => 'states', 'input_type' => 'select'])->update(['options_list' => 'gadm-level-1']);
+        FormQuestion::whereIn('linked_field_key', ['org-level-2-past-restoration', 'pro-pit-level-2-proposed'])
+            ->update(['input_type' => 'select', 'multichoice' => true, 'options_list' => 'gadm-level-2']);
+
+        $this->info("\n\nUpdating user countries...");
+        foreach (User::whereNotNull('country')->get() as $user) {
+            $user->update(['country' => $this->findCountry($user->country)]);
+        }
+
+        $this->info("\n\nUpdating leadership nationalities...");
+        foreach (Leaderships::whereNotNull('nationality')->get() as $leadership) {
+            $leadership->update(['nationality' => $this->findCountry($leadership->nationality)]);
+        }
+    }
+
+    protected function handleMigrationSet($query, $definitions)
+    {
+        $count = (clone $query)->count();
+        $this->withProgressBar($count, function ($progressBar) use ($query, $definitions, &$chunks) {
+            $query->chunk(100, function ($chunk) use ($progressBar, $definitions, &$chunks) {
+                foreach ($chunk as $entity) {
+                    foreach ($definitions as $type => $columns) {
+                        foreach ($columns as $column) {
+                            if (Str::endsWith($type, '_single')) {
+                                $values = [$entity[$column]];
+                            } else {
+                                $values = $entity[$column];
+                            }
+
+                            if (! empty($values)) {
+                                if (Str::startsWith($type, 'gadm_0')) {
+                                    $values = collect($values)->filter()->map(
+                                        fn ($country) => $this->findCountry($country)
+                                    )->filter()->toArray();
+                                } else {
+                                    $values = collect($values)->filter()->map(
+                                        fn ($state) => $this->findState($state)
+                                    )->filter()->toArray();
+                                }
+                            }
+
+                            if (Str::endsWith($type, '_single')) {
+                                $entity[$column] = data_get($values, 0);
+                            } else {
+                                $entity[$column] = $values;
+                            }
+                        }
+                    }
```

Comment:
There is too many if conditions in this block, I can see some small improvements, you can store the _single and gadm_0 in to variables to use to validate below. I think this snippet improves the readability and maintainability.

```suggestion
                    $isSingle = Str::endsWith($type, '_single');
                    $isCountry = Str::startsWith($type, 'gadm_0');

                    foreach ($columns as $column) {
                        $values = $isSingle ? [$entity[$column]] : $entity[$column];
                        
                        $values = collect($values)
                            ->map(fn ($v) => $isCountry ? $this->findCountry($v) : $this->findState($v))
                            ->filter()
                            ->toArray();

                        $entity[$column] = $isSingle ? data_get($values, 0) : $values;
                    }
```

Reply:
I'm going to apply this suggestion. However, just a note for the future: I typically don't get fussy over the formatting, readability or maintainability of one-off scripts. They don't need to be maintainable, they only get run once. In fact, I should pull together a tech debt ticket soon to clear out some old ones. In this particular case, I tested the heck out of this code myself, and with a refactor, while it looks clean to me, I'm going to have to do some more testing to make sure it still works correctly.

Reply:
I'm actually going to need to roll this back. The formatting wasn't right, although I don't understand why from this diff, and the the script didn't work correctly anymore. I don't think it's worth spending the time to try to figure out why when the previous version has already been thoroughly tested.